test/models/publisher_test.rb

```diff
@@ -17,6 +17,42 @@ class PublisherTest < ActiveSupport::TestCase
     Rails.application.secrets[:api_eyeshade_offline] = @prev_offline
   end
 
+  describe "#confirm_pending_email!" do
+    let(:publisher) { publishers(:default) }
+    let(:confirm_email) { nil }
+
+    describe "confirm_email" do
+      describe "pending_email" do
+        describe "!email" do
+          it "should set email to pending_email" do
+          end
+        end
+
+        describe "confirm_email == pending_email" do
+          it "should set email to pending_email" do
+          end
+
+          it "should set pending_email to nil" do
```

Comment:
Not sure if we want to have that in this way. Maybe we want to set it to nil only when the email is changed,
otherwise, other auth logins will be greedy and delete the pending_email. It's not a super huge issue anyway and I would not change this unless it's needed.